nemo/collections/asr/parts/submodules/multi_head_attention.py

```diff
@@ -256,6 +259,250 @@ def forward(self, query, key, value, mask, pos_emb, cache=None, cache_next=None)
         return out
 
 
+class RelPositionMultiHeadAttentionLocal(RelPositionMultiHeadAttention):
+    """Multi-Head Attention layer of Transformer-XL with sliding window local attention.
+    Paper: https://arxiv.org/abs/1901.02860
+    Args:
+        n_head (int): number of heads
+        n_feat (int): size of the features
+        dropout_rate (float): dropout rate
+        pos_bias_u (Tensor): the positional bias matrix U
+        pos_bias_v (Tensor): the positional bias matrix V
+        att_context_size (
+        max_cache_len (int): the maximum size of cache
+    """
+
+    def __init__(self, n_head, n_feat, dropout_rate, pos_bias_u, pos_bias_v, att_context_size, max_cache_len=0):
+        """Construct an RelPositionMultiHeadedAttention object."""
+        super().__init__(
+            n_head=n_head,
+            n_feat=n_feat,
+            dropout_rate=dropout_rate,
+            pos_bias_u=pos_bias_u,
+            pos_bias_v=pos_bias_v,
+            max_cache_len=max_cache_len,
+        )
+        self.att_context_size = att_context_size
+
+    # Longformer implementation for no overlap case adapted for arbitrary left and right chunk size
```

Comment:
Is it the overlapping or the non-overlapping one?
Overlapping one gave better accuracy and the same output with RelPositionMultiHeadedAttention right?
Shouldn't we add the one instead?
Also if we plan to add the other version in future, we may need to mention the overlapping or non-overlapping in the names and docs.

Reply:
This is overlapping, fixed some of the comments. Since this is default in longformer and I'm not sure if we need to add non-overlapping for now I thought to not add it to names for simplicity